testsuite compilation with ghc-9.12.2

Stackage Nightly with ghc-9.12.2:

```
       /home/curators/work/unpack-dir/unpacked/store-0.7.20-29b0a25d2b9eef3b06df7f67c4dc56a81157e1d9e6f262ff890fbe5ac017414c/test/Da
ta/StoreSpec.hs:347:10: error: [GHC-39999]
           • No instance for ‘Serial IO Type’
               arising from a use of ‘smallcheck-1.2.1.1:Test.SmallCheck.Property.changeDepth’
           • In the first argument of ‘($)’, namely
               ‘smallcheck-1.2.1.1:Test.SmallCheck.Property.changeDepth
                  (\ _ -> 2)’
             In the second argument of ‘($)’, namely
               ‘(smallcheck-1.2.1.1:Test.SmallCheck.Property.changeDepth
                   (\ _ -> 2)
                   $ (\ x_ahdO -> checkRoundtrip False (x_ahdO :: Type)))’
             In the second argument of ‘($)’, namely
               ‘(Test.Hspec.SmallCheck.property
                   $ (smallcheck-1.2.1.1:Test.SmallCheck.Property.changeDepth
                        (\ _ -> 2)
                        $ (\ x_ahdO -> checkRoundtrip False (x_ahdO :: Type))))’
           |
       347 |         $(do insts <- getAllInstanceTypes1 ''Store
           |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^... 
```

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -1,5 +1,17 @@
 # ChangeLog
 
+## Unreleased
+
+* Fix test compilation on GHC 9.10+, where `template-haskell` types
+  (`Type`, `TySynEqn`, `PkgName`, etc.) live in `GHC.Internal.TH.Syntax`
+  rather than `Language.Haskell.TH.Syntax`. The `isThName` filter that
+  skips TH AST types from the auto-generated `Store` roundtrip enumeration
+  was matching only the old module name and silently let the TH types
+  through, causing `No instance for 'Serial IO Type'` style errors.
+  See [#182][].
+
+[#182]: https://github.com/mgsloan/store/issues/182
+
 ## 0.7.20
 
 * Fixes build of test with `vector-0.13.2.0`. See [#181][].

--- a/test/Data/StoreSpec.hs
+++ b/test/Data/StoreSpec.hs
@@ -391,8 +391,16 @@ spec = do
              omitTys <- (omitTys0 ++) <$> mapM (\ty -> [t| PV.Vector $(pure ty) |]) omitTys0
              let f ty = isMonoType ty && ty `notElem` omitTys && null (listify isThName ty)
                  filtered = filter f insts
-                 -- Roundtrip testing of TH instances is disabled - see issue #150
-                 isThName n = nameModule n == Just "Language.Haskell.TH.Syntax"
+                 -- Roundtrip testing of TH instances is disabled - see issue #150.
+                 -- GHC 9.10+ ships template-haskell inside `ghc-internal`, so
+                 -- `nameModule` returns `GHC.Internal.TH.Syntax` for TH AST
+                 -- types (Type, TySynEqn, PkgName, SourceUnpackedness, ...).
+                 -- Match both module names so the filter keeps working on both
+                 -- old and new GHCs. See issue #182.
+                 isThName n = nameModule n `elem`
+                              [ Just "Language.Haskell.TH.Syntax"
+                              , Just "GHC.Internal.TH.Syntax"
+                              ]
              smallcheckManyStore verbose 2 $ map return filtered)
     it "Store on non-numeric Float/Double values" $ do
         let testNonNumeric :: forall a m. (RealFloat a, Eq a, Show a, Typeable a, Store a, Monad m, MonadFail m) => Proxy a -> m ()

